lib/dialects/sqlite3/transaction.js

```diff
@@ -0,0 +1,18 @@
+const Transaction = require('../../execution/transaction');
+
+class Transaction_Sqlite extends Transaction {
+  begin(conn) {
+    // SQLite doesn't really support isolation levels, it is serializable by
+    // default and so we override it to ignore isolation level.
+    // There is a `PRAGMA read_uncommitted = true;`, but that's probably not
+    // what the user wants
+    if (this.isolationLevel) {
+      throw new Error(
```

Comment:
Having slept on this, I think warning that the isolation level will be ignored would be better, since SQLite is often used as a stand-in for a real DB, and having "if (test_env)" conditions in code would be a huge hassle.
I think transaction has reference to the client, so it should be possible to do `this.client.logger.warn`.